mautic/mautic
# TPROD-336 - FocusBundle - Convert PHP Templates to Twig Templates

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. 



- `/s/focus` List
  - renders correctly (with and without focus items)
  - can be sorted and filtered
- `/s/focus/new` New
  - Form is rendered correctly
  - Builder works as expected
- `/s/focus/edit/{id}` Edit
  - Form is rendered correctly
  - Builder works as expected
- `/s/focus/view/{id}` Details
  - page is rendered correctly
- Focus Item JS can be placed onto a web page and is rendered correctly (can be done using Landing Pages within Mautic. Best to use HTML with little to no other elements)
- Create/Edit a Campaign > Open Builder > Add Decision for Visits Page > Add a new action for the yes branch > Select **Show Focus Item** > Make sure form renders

**NOTES**
- If you find errors, let me know how to setup the Focus Item the exact same. Try to toggle different settings on/off to see if a setting is on/off will make a difference

**KNOWN ISSUES** that exist on the 5.x branch
- GIVEN you are on `/s/focus` AND have existing Focus Items WHEN you click on the publish toggle to publish or unpublish THEN it will only work once. You will need to reload the page in order to get this to work again
